src/uwtools/drivers/forecast.py

```diff
@@ -138,82 +132,60 @@ def requirements(self) -> None:
         ???
         """
 
-    def resources(self, platform: dict) -> Mapping:
+    def resources(self) -> Mapping:
         """
         Parses the config and returns a formatted dictionary for the batch script.
         """
-        # Add required fields to platform.
-        # Currently supporting only slurm scheduler.
+
         return {
-            "account": platform["account"],
-            "nodes": 1,
-            "queue": "batch",
-            "scheduler": self.job_scheduler,
-            "tasks_per_node": 1,
-            "walltime": "00:01:00",
+            "account": self._experiment_config["user"]["account"],
+            "scheduler": self._experiment_config["platform"]["scheduler"],
+            **self._config["jobinfo"],
```

Comment:
Do you mean, should we have a kind of dummy entry in YAML that doesn't actually provide the value, but is a pointer to a value somewhere else? If so, I don't think so: I think we should have a single, sensible place for each value, and then just read from those locations as needed -- like what you have here. Maybe we'll eventually want to factor that out, when we have more use cases to guide us, so that we're not doing the same lookups in many places, places that will all have to be updated if something about the structure of the config changes. That could start to involve distilling collections of YAML-derived facts into dataclass config objects -- instances of real _classes_ that we'd define, not just more `dict` objects. I think that would achieve a goal similar to "add[ing] account and scheduler to the `jobinfo` section", except that it would be done programmatically, in Python, with all the quality assurance tools available to us there, instead of in YAML.

Reply:
I was suggesting that approach, yes. I appreciate your argument here and it's exactly what I was on the fence about.